test/python/synthesis/test_synthesis.py

```diff
@@ -141,24 +130,10 @@ def check_one_qubit_euler_angles(self, operator, basis="U3", tolerance=1e-14, si
             np.abs(maxdist) < tolerance, f"Operator {operator}: Worst distance {maxdist}"
         )
 
-    @contextlib.contextmanager
-    def assertDebugOnly(self):  # FIXME: when at python 3.10+ replace with assertNoLogs
-        """Context manager, asserts log is emitted at level DEBUG but no higher"""
-        with self.assertLogs("qiskit.synthesis", "DEBUG") as ctx:
-            yield
-        for i in range(len(ctx.records)):
-            self.assertLessEqual(
-                ctx.records[i].levelno,
-                logging.DEBUG,
-                msg=f"Unexpected logging entry: {ctx.output[i]}",
-            )
-            self.assertIn("Requested fidelity:", ctx.records[i].getMessage())
```

Comment:
Removing the debug logging does result in a little less-complete test coverage because now test suite is notchecking that `calculated_fidelity == actual_fidelity` in as many places. Could restore this by adding more assertAlmostEqual(calculated_fidelity, actual_fidelity) in the spots where `assertDebugOnly` was previously, or by moving the debug logging into the python __init__

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11946/commits/03c4af53523f4fabbb72d82fe4a0bea61d09ecc4